Fixed configuration-reference.md for cache staleness in the Spark con… (#33078)
Azure/azure-sdk-for-java: sdk/cosmos/azure-cosmos-spark_3_2-12/docs/configuration-reference.md

| `spark.cosmos.read.maxItemCount`                               | `1000`  | Overrides the maximum number of documents that can be returned for a single query- or change feed request. The default value is `1000` - consider increasing this only for average document sizes significantly smaller than 1KB or when projection reduces the number of properties selected in queries significantly (like when only selecting "id" of documents etc.).                                                                                                                                                                                                                                                                                         |

> | `spark.cosmos.read.maxIntegratedCacheStalenessInMS`        | None    | Sets the max time window in milliseconds for which point read results or query results remain cached in the integrated cache in the dedicated gateway connectivity mode. Learn more about `MaxIntegratedCacheStaleness` [here](https://learn.microsoft.com/azure/cosmos-db/integrated-cache)                                                                                                                                                                                                                                                                                                                                                                      |

Reviewer: Looks good but I have a suggested edit

```suggestion
| `spark.cosmos.read.maxIntegratedCacheStalenessInMS`        | None    | Sets the max staleness window in milliseconds 
 for the point read or query request results in the integrated cache when using the dedicated gateway. Learn more about `MaxIntegratedCacheStaleness` [here](https://learn.microsoft.com/azure/cosmos-db/integrated-cache#maxintegratedcachestaleness)                                                                                                                                                                                                                                                                                                                                                                      |
```

Author: Thanks. I've gone ahead and committed it.
